src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/MethodColdCodeNode.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using Internal.Text;
+using Internal.TypeSystem;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    public class MethodColdCodeNode : ObjectNode, ISymbolDefinitionNode
+    {
+        private ObjectData _methodColdCode;
+        private MethodDesc _owningMethod;
+
+        public MethodColdCodeNode(MethodDesc owningMethod)
+        {
+            _owningMethod = owningMethod;
+        }
+
+        public int Offset => 0;
+
+        public override ObjectNodeSection Section
+        {
+            get
+            {
+                // TODO, Unix
+                return ObjectNodeSection.ManagedCodeWindowsContentSection;
+            }
+        }
+
+        public override bool IsShareable => false;
+
+        // This ClassCode must be larger than that of MethodCodeNode to ensure it got sorted at the end of the code
```

Comment:
ClassCode should be chosen by a fair dice roll. There are no guarantees on ordering and this will be fragile/subtle.

I would suggest creating a new ObjectNodeSection for it and have the PE emitter emit it however it sees fit (at the end of the physical .text section?).

It's similar how this is done for native code: https://devblogs.microsoft.com/oldnewthing/20181107-00/?p=100155 where we would probably use `.text` for hot code and `.text$cold` for cold code.

The PE emitter can apply additional sorting/policies within the section.

Reply:
I filed #1912 to track this issue, meanwhile, would you mind if I merge this in first to unblock the team? 
I am trying to get this feature branch ready so that our interns can start work on top of it.

Reply:
Didn't realize this is runtimelab (I got pinged because of the codeowners file). This is good enough for runtimelab for sure

Reply:
I concur with Michal that a new ObjectNode section should be used for cold code (not sure whether it needs to be Windows / Unix-specific, I don't know the rationale for the pre-existing choice regarding hot code) and ReadyToRunObjectWriter should decode it to a new PE code section to be put after the (hot) code, see here:

https://github.com/dotnet/runtime/blob/8558cdc3e26a0e28c386168be519f818678aadef/src/coreclr/tools/aot/ILCompiler.ReadyToRun/ObjectWriter/R2RPEBuilder.cs#L253

Alternatively (if there's pushback towards generating two code sections as indicated by the code comment at line 256) we'll probably need to tweak the code around

https://github.com/dotnet/runtime/blob/8558cdc3e26a0e28c386168be519f818678aadef/src/coreclr/tools/aot/ILCompiler.ReadyToRun/CodeGen/ReadyToRunObjectWriter.cs#L301

to perhaps stash cold code sections instead of sending them to the section writer right away and flush them after all hot code has been processed.